### ISSUE TITLE
Helm setValueTemplates IMAGE_NAME not reflecting artifactOverrides selection

### ISSUE THREAD

**Issue by mentos1386** [GoogleContainerTools/skaffold#5317]
### Expected behavior

Expecting that `setValueTemplates` for Helm Delpoyment `{{.IMAGE_REPO}}` and `{{.IMAGE_TAG}}` would reflect values for image set with `artifactOverrides`. This was introduced with the following PR https://github.com/GoogleContainerTools/skaffold/pull/4278.

### Actual behavior

It reflects the values of first image in `build.artifacts` array. If we want to actualy access the `second-image` artifact.
We have to use `{{.IMAGE_REPO2}}` and `{{.IMAGE_TAG2}}`.

```
time="2021-01-28T13:24:51+01:00" level=debug msg="EnvVarMap: map[
  DIGEST:first-image:d1507d162d85321f449a1467b87df2bc2143b8a9431abfdf4f01c6097ba1c522
  DIGEST2:second-image:d8f50efad22d80871c4a788cb7ebf0571dbbe50b648d5740152056b411ea267e
  DIGEST_ALGO:first-image
  DIGEST_ALGO2:second-image
  DIGEST_HEX:d1507d162d85321f449a1467b87df2bc2143b8a9431abfdf4f01c6097ba1c522
  DIGEST_HEX2:d8f50efad22d80871c4a788cb7ebf0571dbbe50b648d5740152056b411ea267e
  IMAGE_DIGEST:
  IMAGE_DIGEST2:
  IMAGE_NAME:first-image
  IMAGE_NAME2:second-image
  IMAGE_REPO:first-image
  IMAGE_REPO2:second-image
  IMAGE_TAG:d1507d162d85321f449a1467b87df2bc2143b8a9431abfdf4f01c6097ba1c522
  IMAGE_TAG2:d8f50efad22d80871c4a788cb7ebf0571dbbe50b648d5740152056b411ea267e
]"
```

It's good to have access to other images, but i would preffer to not have to relay on order of artifacts to pick the correct value. Especially with "multi-config" support now added, when order can be changed by providing `-m one,two,four`, different subset of modules.

### Information

- Skaffold version: `v1.18.0`
- Operating system: `Linux 5.10.7-3-MANJARO #1 SMP PREEMPT Fri Jan 15 21:11:34 UTC 2021 x86_64 GNU/Linux`
- Contents of skaffold.yaml:
```
build:
  artifacts:
    - image: first-image
      context: first
      docker:
        dockerfile: Dockerfile
    - image: second-image
      context: second
      docker:
        dockerfile: Dockerfile

deploy:
  helm:
    releases:
      - name: example
        chartPath: example
        setValueTemplates:
          "image.name": "{{.IMAGE_NAME}}"
          "image.tag": "{{.IMAGE_TAG}}"
        artifactOverrides:
          image: second-image
```

### Steps to reproduce the behavior

// TODO: Will create a repo asap

1. a clonable repository with the sample skaffold project
2. `skaffold dev -v debug`
3. `helm get values example` will show `image.name` as `first-image` instead of `second-image`

**Comment by nkubala** [comment]
@mentos1386 thanks for the issue. the least we could do here is improve our documentation on how these values are surfaced - this is tricky though, since `{{.IMAGE_NAME}}` is really a singular thing. i'm not sure the best way we should handle surfacing multiple images apart from appending numbers to the end of the variable name (what we're doing now). what would you like to see instead?

**Comment by mentos1386** [comment]
@nkubala Issue is that `{{.IMAGE_NAME}}` is not deterministic in scope of one `kind: Config` as it's value is dependent on order of how this config is included/merged/required by others.

Which means you cannot safely use it, without taking in to the account the whole scope of all configs that required for some action.

This makes usage with micro service architecture hard, if not impossible. When services have their own artifacts defined.

I would say that it should be changed in one of two ways:
1.  To maintain compatibility with accessing other artifacts, it could be available under `.IMAGE_NAME_<artifact.image>`. As a "map" in a sense. This would allow me to do `.IMAGE_NAME_first-image` which would always output the name for `first-name` artifact, regardless of order.
2. Other option, which would abandon availability of other artifacts, would be to just make `.IMAGE_NAME` always point to the artifact that is defined in `artifactOverrides`. There would be no other `.IMAGE_NAME<index>` at this point.

**Comment by tejal29** [comment]
Triage discussion notes:
1) We are looking into first option as a solution,

**Comment by iosifnicolae2** [comment]
We're experiecing the same problem which makes skaffold v2.0.1 unusable(https://github.com/GoogleContainerTools/skaffold/issues/8008)

**Comment by aaron-prindle** [comment]
Dupe of #6207 
This is fixed now with #8175 

This was recently added, the format is:
`IMAGE_[NAME|TAG|REPO|DOMAIN|REPO_NO_DOMAIN]_<artifact-name>`
